packages/react/src/components/Video/VideoGif.tsx

```diff
@@ -0,0 +1,136 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import {
+  createShorthandFactory,
+  UIComponent,
+  UIComponentProps,
+  commonPropTypes,
+  applyAccessibilityKeyHandlers,
+} from '../../lib'
+import { videoGifBehavior } from '../../lib/accessibility'
+import { Accessibility, AccessibilityActionHandlers } from '../../lib/accessibility/types'
+import Image from '../Image/Image'
+import Video from '../Video/Video'
+import * as _ from 'lodash'
+
+import { ReactProps } from '../../types'
+
+export interface VideoGifProps extends UIComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default videoGifBehavior
+   * */
+  accessibility?: Accessibility
+
+  /** Video source URL. */
+  src?: string
+
+  /** Image source URL for when video isn't playing */
+  poster?: string
+
+  /** whether the gif should be playing */
+  playing?: boolean
+}
+
+export interface VideoGifState {
+  isPlaying: boolean
+}
+
+/**
+ * A GIF is a muted segment of a video
+ * @accessibility
+ * If GIF should be visible to screen readers, textual representation needs to be provided in 'alt' or 'title' property.
+ *
+ * Other considerations:
+ *  - when alt and title property are empty, then Narrator in scan mode navigates to the gif and narrates it as empty paragraph
+ */
+class VideoGif extends UIComponent<ReactProps<VideoGifProps>, VideoGifState> {
+  static create: Function
+
+  static className = 'ui-video-gif'
+
+  static displayName = 'VideoGif'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      children: false,
+      content: false,
+    }),
+    playing: PropTypes.bool,
+    poster: PropTypes.string,
+    src: PropTypes.string,
+  }
+
+  static defaultProps = {
+    as: 'span',
+    accessibility: videoGifBehavior as Accessibility,
+  }
+
+  public state = {
+    isPlaying: this.props.playing,
+  }
+
+  shouldComponentUpdate(nextProps: VideoGifProps, nextState: VideoGifState): boolean {
```

Comment:
![image](https://user-images.githubusercontent.com/14183168/55156258-1dcb0500-5162-11e9-92a6-c2c58e4572c9.png)

---

We have `AutoControlledComponent` for this 😺 

1. Remove `shouldComponentUpdate()` at all
2. Use `AutoControlledComponent`:

```diff
-class VideoGif extends UIComponent<ReactProps<VideoGifProps>, VideoGifState> {
+class VideoGif extends AutoControlledComponent<ReactProps<VideoGifProps>, VideoGifState> {
```
3. Add `autoControlledProps`:
```
static autoControlledProps = ['playing']
```
4. Add `defaultPlaying` to `VideoGifProps` and `propTypes`
5. Use `playing` from `this.state` 🚀  and replace `this.setState({ playing: X })` with `this.trySetState({ playing: X })`

Reply:
Interesting, I'm only worried that won't work for the case where the video state is playing and the component parent changes the defaultPlaying prop to false. In that case, we'd want to set the playing state to false. That's really the only reason I implemented shouldComponentUpdate

Reply:
Change to auto controlled